src/imgui/ImGuiDebugger.cc

```diff
@@ -240,24 +240,22 @@ void ImGuiDebugger::showMenu(MSXMotherBoard* motherBoard)
 		ImGui::Separator();
 		auto getBreakpointFiles = [] {
 			std::vector<std::string> names;
-			for (auto context = userDataFileContext(BREAKPOINT_DIR);
-			     const auto& path : context.getPaths()) {
-				foreach_file(path, [&](const std::string& fullName) {
+			foreach_file(FileOperations::join(openmsx::FileOperations::getUserOpenMSXDir(), "breakpoints"),
+				[&](const std::string& fullName) {
 					if (fullName.ends_with(".breakpoints")) {
 						names.emplace_back(fullName);
 					}
 				});
-			}
 			std::ranges::sort(names, StringOp::caseless{});
 			return names;
 		};
 		im::Menu("Breakpoints", [&]{
 			auto listFiles = [&](const std::vector<std::string>& names) {
 				std::optional<std::string> selectedFile;
 				im::ListBox("##select-bp", [&]{
-					for (const auto& name : names) {
-						auto displayName = std::string(FileOperations::stripExtension(FileOperations::getFilename(name)));
-						if (ImGui::Selectable(displayName.c_str())) {
+					for (const auto& [id, name] : enumerate(names)) {
+						if (auto displayName = std::string(FileOperations::stripExtension(FileOperations::getFilename(name)));
+						    ImGui::Selectable(tmpStrCat(displayName, "##", id).c_str())) {
```

Comment:
That certainly works. And I think we still use this approach in various other locations in openMSX.

But in new code I prefer the approach with `ImGui::PushID()` and `ImGui::PopID()`, or better our c++ wrappers around that e.g. `im::ID(int)`.

For the combination of a for-loop and in each iteration a different ID, we also have `im::ListClipperID()`. Maybe not super important for this specific case, but that immediately also optimized drawing lists with many items (where only part of the items are visible at the same time).